src/Forms/FormField.php

```diff
@@ -1221,30 +1240,30 @@ public function Type()
     }
 
     /**
-     * Utility method to call an extension hook which allows the result of validate() calls to be adjusted
-     *
-     * @param bool $result
-     * @param Validator $validator
-     * @return bool
+     * Determines whether the field is valid or not based on its value
      */
-    protected function extendValidationResult(bool $result, Validator $validator): bool
```

Comment:
Yes I did look at it. That's no longer compatible with the new way of doing this

The old `FormField::validate()` method had a signature of `validate(Validator $validator): bool`, whereas this has now been updated to match DataObject and DBField with `validate(): ValidationResult`

The replicate the extension functionality, there's a new extension hook `updateValidate` on FormField::validate() and a `ValidationResult` param to use. I've used this in combination with `$this->beforeExtending('updateValidate')` and calling parent::validate() on any overridden validate method so that the extension hook is always called.  I've added a unit test on FormFieldTest::testUpdateValidateHookCalled() to ensure this is happening

Reply:
Sweet. I think so long as that test is still there that's the main thing.

I do like that this now means the `updateValidate` extension hook is the same method signature for both `DataObject` and `FormField` - we should call that out in the changelog probably.

Reply:
Yup already in the 6.0.0 changelog - https://github.com/silverstripe/developer-docs/pull/621/files#diff-027bda823f29903fb90f1ca3e86dc918c3b6a814c50096c66a3074b1da24eb62R797

Reply:
Have mentioned in 6.0.0 changelog - https://github.com/silverstripe/developer-docs/pull/621/files#diff-027bda823f29903fb90f1ca3e86dc918c3b6a814c50096c66a3074b1da24eb62R799